src/dotnet/ShellShim/ShellShimRepository.cs

```diff
@@ -138,18 +125,44 @@ public void RemoveShim(string commandName)
                 });
         }
 
-        internal void CreateConfigFile(FilePath outputPath, FilePath entryPoint, string runner)
+        private void CreateApphostShimAndConfigFile(string commandName, FilePath entryPoint)
         {
-            XDocument config;
-            using (var resource = typeof(ShellShimRepository).Assembly.GetManifestResourceStream(LauncherConfigResourceName))
+            string appHostSourcePath;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                config = XDocument.Load(resource);
+                appHostSourcePath = Path.Combine(_appHostSourceDirectory, ApphostNameWithoutExtension + ".exe");
             }
+            else
+            {
+                appHostSourcePath = Path.Combine(_appHostSourceDirectory, ApphostNameWithoutExtension);
+            }
+
+            var ModifiedAppHostPath = EmbedAppNameInHost.EmbedAndReturnModifiedAppHostPath(
+                appHostSourcePath: appHostSourcePath,
+                appHostDestinationDirectoryPath: _shimsDirectory.Value,
+                appBinaryName: Path.GetFileName(entryPoint.Value));
+
+            var config = JsonConvert.SerializeObject(
+                new RootObject
+                {
+                    startupOptions = new StartupOptions
+                    {
+                        appRoot = entryPoint.GetDirectoryPath().Value
+                    }
+                });
 
-            var appSettings = config.Descendants("appSettings").First();
-            appSettings.Add(new XElement("add", new XAttribute("key", "entryPoint"), new XAttribute("value", entryPoint.Value)));
-            appSettings.Add(new XElement("add", new XAttribute("key", "runner"), new XAttribute("value", runner ?? string.Empty)));
-            config.Save(outputPath.Value);
+            File.WriteAllText(GetConfigPath(commandName).Value, config);
+            File.Move(ModifiedAppHostPath, GetShimPath(commandName).Value);
```

Comment:
It seems like an unnecessary `File.Move` operation when `EmbedAndReturnModifiedAppHostPath` could simply take the destination path to write the embedded app host to?

Reply:
it is the destinationDirectory. It will use dll's name by default

Reply:
Right, but why create the file and then rename it when it could just create the file at the right location?

Reply:
I want to keep the 2 copy of the file (another on SDK) code the same as much as possible. I don't want to further edit the code other than convert it to a method.

Reply:
This code isn't exactly complicated nor is there very much of it; I don't think it would be burdensome to maintain a change that prevents another operation that can fail.

Regardless, what happens right now if there exists a shim with the same name as the dll we're about to create a shim for?  It appears that it will be truncated given this "write as <dll_name> and rename to command" process.  We should be writing it as the desired file.

Reply:
i see. the following is a problem. I will change the api